auacm/app/modules/competition_manager/views.py

```diff
@@ -181,3 +202,129 @@ def get_competition_data(cid):
+        session.add(CompUser(
+            cid=cid,
+            username=user.username,
+            team=user.display
+        ))
+        socketio.emit('new_user', {
+            'cid': cid,
+            'user': {
+                'display': user.display,
+                'username': user.username
+            }
+        },
+        namespace='/register')
+    session.flush()
+    session.commit()
+
+    return serve_response({})
+
+
+@app.route('/api/competitions/<int:cid>/unregister', methods=['POST'])
+@login_required
+def unregister_for_competition(cid):
+    """ Called when a user wants to register for a competition.
+
+    All the user has to do is submit a post to this url with no form data.
+    From their logged-in status, we'll go ahead and add them to the competiton
+    as an individual (team name is default their display name).
+    """
+    if session.query(Competition).filter(Competition.cid == cid).first() \
+            is None:
+        return serve_error('Competition does not exist', response_code=404)
+
+    if False and current_user.admin == 1 and 'users' in request.data:
+        registrants = loads(request.data['users'])
+    else:
+        registrants = list()
+        registrants.append(current_user.username)
+
+    for user in registrants:
+        session.query(CompUser)\
+                .filter(CompUser.username == user,
+                CompUser.cid == cid)\
+                .delete()
+    session.flush()
+    session.commit()
+
+    return serve_response({})
+
+
+@app.route('/api/competitions/<int:cid>/teams', methods=['GET'])
+@login_required
+def get_competition_teams(cid):
+    comp_users = session.query(CompUser, User).join(User,
+            User.username == CompUser.username).filter(CompUser.cid == cid)\
+            .all()
+
+    teams = dict()
+    for user in comp_users:
+        if user.CompUser.team not in teams:
+            teams[user.CompUser.team] = list()
+        teams[user.CompUser.team].append({
+            'username': user.User.username,
+            'display': user.User.display
+        })
+
+    return serve_response(teams)
+
+
+@app.route('/api/competitions/<int:cid>/teams', methods=['PUT'])
+@login_required
+def put_competition_teams(cid):
+    if current_user.admin == 0:
+        return serve_error('You must be an admin to update teams',
+                response_code=401)
+
+    try:
+        teams = loads(request.form['teams'])
+    except KeyError as err:
+        return serve_error('You must include the parameter \'teams\'.',
+                response_code=400)
+
+    # Delete all of the old CompUser rows for this competition
+    session.query(CompUser).filter(CompUser.cid == cid).delete()
+
+    for team in teams:
+        for user in teams[team]:
+            session.add(CompUser(
+                cid=cid,
+                username=user,
+                team=team
+            ))
+
+    session.flush()
+    session.commit()
+
+    return serve_response({})
```

Comment:
If we returned the teams, we'd just be returning the JSON data that the user just specified in the PUT. It would be kind of redundant. All we need to know here is that the request was successful.